internal/controller/capacity.go

```diff
@@ -195,6 +195,38 @@ func (*InstasliceReconciler) getStartIndexFromPreparedState(instaslice *inferenc
 	return newStart
 }
 
+// getTotalFitForProfileOnGPU checks how many times a profile can fit on a given GPU.
+func (r *InstasliceReconciler) getTotalFitForProfileOnGPU(instasliceObj *inferencev1alpha1.Instaslice, profileName string, remaining int32) int32 {
```

Comment:
Can we not modify the `getStartIndexFromPreparedState` and report all the valid profile after selecting one valid profile for a pod?

Reply:
We can, but it adds more code logic and I believe it overcomplicates it a bit. Since I am already calculating the remaining slices for a gpu, reusing that logic with less code makes sense.